File: 0567-permutation-in-string/0567-permutation-in-string.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        int n = s1.length();
        int m = s2.length();
        if(n > m){
            return false;
        }

        int[] temp1 = new int[26];
        int[] temp2 = new int[26];

        for(int i = 0; i<n; i++){
            temp1[s1.charAt(i)-'a']++;
            temp2[s2.charAt(i)-'a']++;
        }

        for(int i = n; i<m; i++){
            if(Arrays.equals(temp1, temp2)){
                return true;
            }
            temp2[s2.charAt(i)-'a']++;
            temp2[s2.charAt(i-n)-'a']--;
        }

        return Arrays.equals(temp1, temp2);
    }
}

File: 0567-permutation-in-string/0567-permutation-in-string.cpp
class Solution {
public:

    bool isSame(int *first, int *second){

        for(int i = 0; i<26; i++){
            if(first[i] != second[i]){
                return false;
            }
        }

        return true;
    }
    bool checkInclusion(string s1, string s2) {
        if(s2.size() < s1.size()){
            return false;
        }
        int first[26] = {0};
        int second[26] = {0};

        for(int i = 0; i<s1.size(); i++){
            first[s1[i]-'a']++;
            second[s2[i]-'a']++;
        }

        int i = 0;

        for(int j = s1.size(); j<s2.size(); j++){
            if(isSame(first, second)){
                return true;
            }
            second[s2[i]-'a']--;
            second[s2[j]-'a']++;
            i++;
        }

        return false;
    }
};